src/pip/_internal/operations/freeze.py

```diff
@@ -60,10 +56,15 @@ def freeze(
     for link in find_links:
         yield '-f {}'.format(link)
     installations = {}  # type: Dict[str, FrozenRequirement]
-    for dist in get_installed_distributions(local_only=local_only,
-                                            skip=(),
-                                            user_only=user_only,
-                                            paths=paths):
+
+    # None is a Falsy value, so pass False if local_only
+    # and user_only is None
+    for dist in get_installed_distributions(
+            local_only=local_only if local_only else False,
+            skip=(),
+            user_only=user_only if user_only else False,
```

Comment:
```suggestion
            local_only=bool(local_only)
            skip=(),
            user_only=bool(user_only),
```

I wonder if we can unify the parameter types of this method and that of `utils.misc.get_installed_distributions` though.

Reply:
Good point about casting to bool, fixed in https://github.com/pypa/pip/pull/8322/commits/e3db5767b8c6a8959dcb5f60b91c41a759fa3d26

> I wonder if we can unify the parameter types of this method and that of utils.misc.get_installed_distributions though.

By this, did you mean something like below, which changes the types from `Optional[bool]` -> `bool` with the default value as False, and not case `local_only` and `user_only` while passing.

```diff
h$ git diff
diff --git a/src/pip/_internal/operations/freeze.py b/src/pip/_internal/operations/freeze.py
index b44d7989..efdde8a4 100644
--- a/src/pip/_internal/operations/freeze.py
+++ b/src/pip/_internal/operations/freeze.py
@@ -42,8 +42,8 @@ logger = logging.getLogger(__name__)
 def freeze(
     requirement=None,  # type: Optional[List[str]]
     find_links=None,  # type: Optional[List[str]]
-    local_only=None,  # type: Optional[bool]
-    user_only=None,  # type: Optional[bool]
+    local_only=False,  # type: bool
+    user_only=False,  # type: bool
```

That's a good idea too, but I am not too sure which way is better

Reply:
Yes, although I haven't really investigate why None was used in the first place.

Reply:
This was done way back in https://github.com/pypa/pip/pull/2173 , but at the call site of [freeze](https://github.com/pypa/pip/blob/master/src/pip/_internal/commands/freeze.py#L101), it seems like `options.local` and `options.user` have [default value `False`](https://github.com/pypa/pip/blob/1ebeab51460f7431a490576e52223fe4b3b22d53/src/pip/_internal/commands/freeze.py#L51-L63)

So maybe the second idea is better than the first.

Reply:
Since there is agreement that second idea is better based on the +1s in the above comment, the behaviour is now updated in https://github.com/pypa/pip/pull/8322/commits/28cd824ccf447effc0d4c80c218f01a858aa55d6